openhands/runtime/remote/runtime.py

```diff
@@ -158,53 +190,40 @@ def __init__(
                 f'{browsergym_arg}'
             ),
             'working_dir': '/openhands/code/',
-            'name': self.container_name,
             'environment': {'DEBUG': 'true'} if self.config.debug else {},
         }
 
-        self.send_status_message('STATUS$WAITING_FOR_CLIENT')
         # Start the sandbox using the /start endpoint
         response = send_request(
             self.session,
             'POST',
-            f'{self.config.sandbox.remote_runtime_api_url}/start',
+            f'{self.config.sandbox.remote_runtime_api_url}/start?id={self.instance_id}',
             json=start_request,
         )
         if response.status_code != 201:
             raise RuntimeError(f'Failed to start sandbox: {response.text}')
-        start_response = response.json()
-        self.runtime_id = start_response['runtime_id']
-        self.runtime_url = start_response['url']
-
+        self._parse_runtime_response(response)
         logger.info(
             f'Sandbox started. Runtime ID: {self.runtime_id}, URL: {self.runtime_url}'
         )
 
+    def _parse_runtime_response(self, response: requests.Response):
+        start_response = response.json()
+        self.runtime_id = start_response['runtime_id']
+        self.runtime_url = start_response['url']
         if 'session_api_key' in start_response:
             self.session.headers.update(
                 {'X-Session-API-Key': start_response['session_api_key']}
             )
-
-        # Initialize the eventstream and env vars
-        super().__init__(
-            config, event_stream, sid, plugins, env_vars, status_message_callback
-        )
-
-        logger.info(
-            f'Runtime initialized with plugins: {[plugin.name for plugin in self.plugins]}'
-        )
-        logger.info(f'Runtime initialized with env vars: {env_vars}')
-        assert (
-            self.runtime_id is not None
-        ), 'Runtime ID is not set. This should never happen.'
-        assert (
-            self.runtime_url is not None
-        ), 'Runtime URL is not set. This should never happen.'
-        self.send_status_message(' ')
+        elif 'sandbox_api_key' in start_response:
+            # FIXME: Remove this once the runtime API is updated
+            self.session.headers.update(
+                {'X-Session-API-Key': start_response['sandbox_api_key']}
+            )
```

Comment:
What's the difference between session/sandbox API key? Can we stick to use one to reduce the cognitive load & simplify the code?

Reply:
yes I misnamed it 🙃 I'm getting rid of sandbox

Reply:
Shall we remove the sandbox_api_key here now since it is already removed? After this i think this PR is good to merge